### PR TITLE
Add patch for filename paths

### PR DESCRIPTION
**Pull request by sevenzing**
#328

### CODE REVIEW

**Comment by rimrakhimov** [comment]
Is the following situation possible with input?

```
"src/@hello/kitty/a.sol": "content",
"src/_hello/kitty/b.sol": "content",
```

**Review by leviska — APPROVED** [review]

**Comment by sevenzing** [comment]
> Is the following situation possible with input?
> 
> ```
> "src/@hello/kitty/a.sol": "content",
> "src/_hello/kitty/b.sol": "content",
> ```

I would say it's not possible, because in solidity you always import it like `import "@openzeppelin/contracts/...";`

I also didn't come up with an idea of how to handle not only first component of path,  however, If you suggest a way how to handle this type of import, I would discuss it.

**Review by rimrakhimov — APPROVED** [review]

File: visualizer/visualizer-server/src/solidity/types.rs [hunk 1]
```diff
@@ -35,12 +46,37 @@ fn output_mask(field_mask: Option<FieldMask>) -> Result<OutputMask, anyhow::Erro
...
+            (path, content)
+        })
+        .collect()
+}
+
+// Should be the same as in
+// https://github.com/ethereum/sourcify/blob/d0882a5d6158d0f56e121835d79860034f072cd8/services/verification/src/services/Injector.ts#L907
+fn sanitize_path(path: &str) -> String {
+    let path = REGEX_ONLY_CHARS.replace_all(path, "_");
+    let path = REGEX_NO_DOTS.replace_all(path.as_ref(), "_");
```
Comment (rimrakhimov) [inline visualizer/visualizer-server/src/solidity/types.rs:70]: Nit: what is the reason behind implementing regex strings as LazyStatic?

**Review by rimrakhimov — APPROVED** [review]

File: visualizer/visualizer-server/src/solidity/types.rs (hunk 1 above)
Comment (sevenzing) [reply visualizer/visualizer-server/src/solidity/types.rs:70]: to be sure that they are compiled when the service is started

**Pull request merged**